services/ui-src/src/components/modals/AddEditReportModal.tsx

```diff
@@ -25,40 +26,68 @@ export const AddEditProgramModal = ({
     useContext(ReportContext);
   const { full_name } = useUser().user ?? {};
   const [submitting, setSubmitting] = useState<boolean>(false);
-  const form: FormJson = formJson;
 
-  const writeProgram = async (formData: any) => {
+  const modalFormJsonMap: any = {
+    MCPAR: mcparFormJson,
+    MLR: mlrFormJson,
+  };
+
+  const modalFormJson = modalFormJsonMap[reportType]!;
+  const form: FormJson = modalFormJson;
+
+  const writeReport = async (formData: any) => {
     setSubmitting(true);
     const submitButton = document.querySelector("[form=" + form.id + "]");
     submitButton?.setAttribute("disabled", "true");
 
-    // prepare payload
-    const programName = formData["programName"];
-    const dueDate = calculateDueDate(formData["reportingPeriodEndDate"]);
-    const combinedData = formData["combinedData"] || false;
-    const reportingPeriodStartDate = convertDateEtToUtc(
-      formData["reportingPeriodStartDate"]
-    );
-    const reportingPeriodEndDate = convertDateEtToUtc(
-      formData["reportingPeriodEndDate"]
-    );
+    let dataToWrite;
+
+    // prepare MCPAR payload
+    if (reportType === "MCPAR") {
```

Comment:
Thought: This if/else is getting large enough that it might start making sense to split it out into separate functions that return the payload. Like `dataToWrite = formMcparPayload()` or something.